charms/slurmctld/src/charm.py

```diff
@@ -127,7 +144,42 @@ def _on_install(self, event: InstallEvent) -> None:
             logger.error(e.message)
             event.defer()
 
-        self._on_write_slurm_conf(event)
+    def _on_start(self, event: StartEvent) -> None:
+        """Set cluster_name and write slurm.conf.
+
+        Note: The start hook can execute multiple times in a charms lifecycle,
+              for example, after a reboot of the underlying instance. This code safeguards
+              against the potentiality of changing the cluster_name in subsequent start hook
+              executions by applying logic that ensures the cluster_name is only set on the
+              first execution of this hook, we log and return on any subsequent start hook
+              event executions.
```

Comment:
[nit]: just a docstring format request:

```suggestion
        Notes: 
            - The start hook can execute multiple times in a charms lifecycle,
              for example, after a reboot of the underlying instance. This code safeguards
              against the potentiality of changing the cluster_name in subsequent start hook
              executions by applying logic that ensures the cluster_name is only set on the
              first execution of this hook, we log and return on any subsequent start hook
              event executions.
```